src/tasks/PageIterator.ts

```diff
@@ -0,0 +1,341 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @module PageIterator
+ */
+
+import {
+  Parsable,
+  RequestAdapter,
+  RequestOption,
+  RequestInformation,
+  HttpMethod,
+  ParsableFactory,
+  ErrorMappings,
+  Headers,
+} from "@microsoft/kiota-abstractions";
+
+/**
+ * Signature representing PageCollection
+ * @property {any[]} value - The collection value
+ * @property {string} [@odata.nextLink] - The nextLink value
+ * @property {string} [@odata.deltaLink] - The deltaLink value
+ * @property {any} Additional - Any number of additional properties (This is to accept the any additional data returned by in the response to the nextLink request)
+ */
+export interface PageCollection<T> {
+  value: T[];
+  odataNextLink?: string;
+  odataDeltaLink?: string;
+  [Key: string]: any;
+}
+
+/**
+ * Signature representing callback for page iterator
+ * @property {Function} callback - The callback function which should return boolean to continue the continue/stop the iteration.
+ */
+export type PageIteratorCallback<T> = (data: T) => boolean;
+
+/**
+ * Signature to define the request options to be sent during request.
+ * The values of the GraphRequestOptions properties are passed to the Graph Request object.
+ * @property {Headers} headers - the header options for the request
+ * @property {RequestOption[]} options - The middleware options for the request
+ */
+export interface PagingRequestOptions {
+  headers?: Headers;
+  requestOption?: RequestOption[];
+}
+
+/**
+ * @enum
+ * Enum representing the state of the iterator
+ */
+export enum PagingState {
```

Comment:
I'd recommend against using enums since they are slowly going away and some ts configurations /deno/etc... now prohibit them. How about a union of string values instead?